src/native/external/zlib-intel.cmake

```diff
@@ -22,4 +22,13 @@ set(ZLIB_SOURCES_BASE
     zutil.c
 )
 
+# Configure our custom allocator
+if(MSVC)
+    set(ZLIB_SOURCES_BASE ${ZLIB_SOURCES_BASE} ../zlib/dotnet_allocator_win.c)
+    add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/DMY_ZCALLOC>)
+else()
+    set(ZLIB_SOURCES_BASE ${ZLIB_SOURCES_BASE} ../zlib/dotnet_allocator_unix.c)
```

Comment:
This is dead code. We only ever use the checked-in zlib sources on Windows.

Reply:
The same comment applies to zlib.cmake.

Reply:
I asked in the group chat and got some guidance on what to put here, so marking this as resolved. If you think other changes still need to be made, please leave another comment! Thanks for finding it. :)